packages/connect-thegraph/src/core/GraphQLWrapper.ts

```diff
@@ -1,30 +1,78 @@
 import fetch from 'isomorphic-unfetch'
-import { Client } from '@urql/core'
+import ws from 'ws'
+import { Client, defaultExchanges, subscriptionExchange, createRequest } from '@urql/core'
+import { SubscriptionClient } from 'subscriptions-transport-ws'
 import { DocumentNode } from 'graphql'
 import { ParseFunction, QueryResult } from '../types'
+import { pipe, subscribe } from 'wonka'
+
+type SubscriptionCallback = (result: any) => void
+type Subscription = {
+  unsubscribe: () => void
+}
 
 export default class GraphQLWrapper {
   #client: Client
   #verbose: boolean
 
   constructor(subgraphUrl: string, verbose = false) {
+    const subscriptionClient = new SubscriptionClient(
+      subgraphUrl.replace('http', 'ws'),
+      {
+        reconnect: true,
+        timeout: 20000
+      },
+      ws
+    )
+
     this.#client = new Client({
       maskTypename: true,
       url: subgraphUrl,
       fetch,
+      exchanges: [
+        ...defaultExchanges,
+        subscriptionExchange({
+          forwardSubscription: (operation: any) => subscriptionClient.request(operation)
```

Comment:
Wonder if `graphql` or `urql` might have support for the operation type instead of using any here.